dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -414,4 +436,88 @@ public Config getConfig() {
+            builder.cache(DockstoreWebserviceApplication.getCache(null));
+            OkHttpClient build = builder.build();
+            ObsoleteUrlFactory obsoleteUrlFactory = new ObsoleteUrlFactory(build);
+            HttpConnector okHttp3Connector = new ImpatientHttpConnector(obsoleteUrlFactory::open);
+            GitHub gitHub = GitHubBuilder.fromEnvironment().withConnector(okHttp3Connector).build();
+
+            GHRepository repository = gitHub.getRepository("dockstore/dockstore-cli");
+            ghRelease = repository.getLatestRelease();
+            allReleases = repository.listReleases();
+        } catch (IOException e) {
+            LOG.info("Could not get CLI releases information from GitHub. ", e);
+            throw new CustomWebApplicationException("Could not get CLI  releases information info from GitHub",
+                    HttpStatus.SC_SERVICE_UNAVAILABLE);
+        }
+        String cliLatestVersion = ghRelease.getName();
+        cliInfo.setCliLatestDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestVersion + "/dockstore");
+        cliInfo.setCliLatestVersion(cliLatestVersion);
+        // Get the latest unstable version
+        // If there is no unstable version return a null as the latest unstable version
+        Optional<String> cliLatestUnstableVersion = getLatestUnstableVersion(allReleases, cliLatestVersion);
+        if (cliLatestUnstableVersion.isPresent()) {
+            cliInfo.setCliLatestUnstableDockstoreScriptDownloadUrl(DOCKSTORE_CLI_RELEASES_URL + "/download/" + cliLatestUnstableVersion.get() + "/dockstore");
+            cliInfo.setCliLatestUnstableVersion(cliLatestUnstableVersion.get());
+        } else {
+            cliInfo.setCliLatestUnstableDockstoreScriptDownloadUrl(null);
+            cliInfo.setCliLatestUnstableVersion(null);
+        }
+        return cliInfo;
+    }
+
+    /**
+     * This method will return the prerelease version that is greater than the latest version if there is one
+     * If there isn't a prerelease version greater than the latest release then don't return one
+     * since we don't want to downgrade to an older untested version
+     *
+     * @param allReleases a PagedIterable of GHRelease objects for releases
+     * @param latestRelease the semantic variable string of the latest release
+     * @return An Optional String of the latest unstable version
+     */
+    private static Optional<String> getLatestUnstableVersion(PagedIterable<GHRelease> allReleases, String latestRelease) {
+        // https://stackoverflow.com/questions/24511052/how-to-convert-an-iterator-to-a-stream
+        Stream<GHRelease> targetStream = StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(allReleases.iterator(), Spliterator.ORDERED), false);
+        return Optional.of(targetStream
+                .filter(GHRelease::isPrerelease)
+                .filter(g -> preReleaseVersionIsGreaterThanLatest(g.getName(), latestRelease))
```

Comment:
I don't think you want to do this here; do it after/if you find the first pre-release.

If the most recent prerelease is older than the current release, the second filter is going to force it to iterate through all releases until it finds one that is both pre-release and newer than the current. So then your fetching every single release from GitHub -- I know it's not that much data, but since you have the lazy stream, might as well not iterate over the whole data set when you don't need to.